### PR TITLE
tarball-ttl: document

### LINKED ISSUE #1115: Please document "tarball-ttl" option
https://github.com/NixOS/nix/blob/master/src/libstore/download.cc#L522 uses a config option called `tarball-ttl`, but I cannot find any documentation for that option in the man pages. I suppose this should be documented in the `nix.conf` man page?

### PR DESCRIPTION
**Pull request by grahamc**
Closes #1115

### CODE REVIEW

**Comment by teto** [comment]
would a mention to fetchGit / fetchurl be too detailed ?

**Comment by grahamc** [comment]
No I don't think so, can you help me out by writing a few sentences about that?

File: doc/manual/command-ref/conf-file.xml [hunk 1]
```diff
@@ -864,6 +864,17 @@ requiredSystemFeatures = [ "kvm" ];
 
   </varlistentry>
 
+  <varlistentry xml:id="conf-tarball-ttl"><term><literal>tarball-ttl</literal></term>
+
+    <listitem>
+      <para>The number of seconds tarballs in
+      <filename>~/.cache/nix/tarballs</filename> are considered fresh.
```
Comment (teto) [inline doc/manual/command-ref/conf-file.xml]: $XDG_CACHE_HOME ?

**Comment by teto** [comment]
as I essentially use it to refresh freshly pushed fixup commits, I would advance sthg like:
`Timeout that triggers builtins.fetchGit to refetch data.`

File: doc/manual/command-ref/conf-file.xml [hunk 2]
```diff
@@ -864,6 +864,26 @@ requiredSystemFeatures = [ "kvm" ];
...
+      are considered fresh.
+      </para>
+
+      <para>This is most relevant for fetching tarballs that change
+      remotely via <envar>NIX_PATH</envar> or <function>fetchGit</function>.
+      </para>
+
+      <para>Setting the TTL to <literal>0</literal> forces Nix to check
+      if the tarball is up to date by ETag if the remote server supports
+      it.</para>
```
Comment (edolstra) [inline doc/manual/command-ref/conf-file.xml]: This suggests that `ETag` is only used if TTL=0, but it's always used if supplied by the server.

Comment (grahamc) [reply doc/manual/command-ref/conf-file.xml]: This is complicated :)

File: doc/manual/command-ref/conf-file.xml [hunk 3]
```diff
@@ -864,6 +864,26 @@ requiredSystemFeatures = [ "kvm" ];
...
 
+  <varlistentry xml:id="conf-tarball-ttl"><term><literal>tarball-ttl</literal></term>
+
+    <listitem>
+
+      <para>Default: <literal>3600</literal> seconds.</para>
+
+      <para>The number of seconds a download is cached in
+      <filename>$XDG_CACHE_HOME/nix/tarballs</filename>
+      are considered fresh.
```
Comment (edolstra) [inline doc/manual/command-ref/conf-file.xml]: `is cached in ... are considered fresh` -> `... is considered fresh`. And maybe add something like "If a cached tarball is older, Nix will check whether it is still up to date using its ETag header, and if not, download the latest version."

**Comment by grahamc** [comment]
Updated!

File: doc/manual/expressions/builtins.xml [hunk 4]
```diff
@@ -347,7 +347,7 @@ stdenv.mkDerivation { … }
     You can change the cache timeout either on the command line with
     <option>--option tarball-ttl <replaceable>number of seconds</replaceable></option> or
     in the Nix configuration file with this option:
-    <literal>tarball-ttl <replaceable>number of seconds to cache</replaceable></literal>.
+    <literal><xref linkend="conf-tarball-ttl" /> <replaceable>number of seconds to cache</replaceable></literal>.
```
Comment (edolstra) [inline doc/manual/expressions/builtins.xml]: Hm, what does this render as?

Comment (grahamc) [reply doc/manual/expressions/builtins.xml]: Screenshot_2019-07-05 Nix Package Manager Guide

It becomes a link, but (at least, locally) doesn't show a special indication that it is clickable. The term is automatically selected as the link text.

**Comment by teto** [comment]
thanks ! happy to see one of the few settings I use documented.

**Pull request merged**